### PR TITLE
Improve CUDA compatibility checks

### PR DESCRIPTION
**Pull request by nevillelyh**
- Check version major/minor instead of exact match
- Suggest compatible version if available

### CODE REVIEW

**Comment by mattt** [comment]
@nevillelyh Rebased on top of #1691 to fix CI failure

File: pkg/config/config.go [hunk 1]
```diff
@@ -461,9 +464,12 @@ Compatible cuDNN version is: %s`,
 				return err
 			}
 			console.Debugf("Setting CUDA to version %s from Torch version", c.Build.CUDA)
-		} else if !slices.ContainsString(torchCUDAs, c.Build.CUDA) {
+		} else if len(slices.FilterString(torchCUDAs, func(torchCUDA string) bool { return version.EqualMinor(torchCUDA, c.Build.CUDA) })) == 0 {
```
Comment (mattt) [inline pkg/config/config.go:467]: I had trouble understanding this conditional. Could we make this an `else` and break this up over a couple lines?

Comment (nevillelyh) [reply pkg/config/config.go]: Fixed!

**Comment by zeke** [comment]
Heads up @marcusmartins in case you're working on something related to this in https://github.com/replicate/cog/pull/1699

> Check that the combination of cuda, python and pytorch is valid before attempting to run/build the image

**Pull request merged**